src/lib.rs

```diff
@@ -233,71 +233,125 @@ impl TransactionMetadata {
+        let now = Instant::now();
+        // Fill in random sender values in the tree
+        for transaction_num in 0..num_transactions {
+            // Get a random index to instantiate values for
+            let s_index = rng.next_u64() as usize % tree_size;
+            s_indices[transaction_num] = s_index;
             let skey = Scalar::random(&mut rng);
-            secret_keys.push(skey);
+            secret_keys[s_index] = skey;
             let pkey = AffinePoint::from(AffinePoint::generator() * skey);
-            let value1 = BaseElement::from(value_elements[i * 2]);
-            let value2 = BaseElement::from(value_elements[i * 2 + 1]);
+            let balance = rng.next_u64();
+            let nonce = rng.next_u64();
             let mut val = [BaseElement::ZERO; AFFINE_POINT_WIDTH + 2];
             val[0..POINT_COORDINATE_WIDTH].copy_from_slice(&pkey.get_x());
             val[POINT_COORDINATE_WIDTH..AFFINE_POINT_WIDTH].copy_from_slice(&pkey.get_y());
-            val[AFFINE_POINT_WIDTH] = value1;
-            val[AFFINE_POINT_WIDTH + 1] = value2;
-            values.push(val);
-            leaves.push(Rescue63::merge(&[
+            val[AFFINE_POINT_WIDTH] = BaseElement::from(balance);
+            val[AFFINE_POINT_WIDTH + 1] = BaseElement::from(nonce);
+            values[s_index] = val;
+            let leaf = Rescue63::merge(&[
                 Hash::new(val[0], val[1], val[2], val[3], val[4], val[5], val[6]),
                 Hash::new(val[7], val[8], val[9], val[10], val[11], val[12], val[13]),
-            ]));
+            ]);
+            // Update the tree with the new leaf
+            tree.update_leaf(s_index, leaf);
         }
-        let mut tree = MerkleTree::<Rescue63>::new(leaves.clone()).unwrap();
         #[cfg(feature = "std")]
         debug!(
-            "Built Merkle tree of depth {} in {} ms",
-            MERKLE_TREE_DEPTH,
+            "Filled in {} sender accounts in {} ms",
+            num_transactions,
             now.elapsed().as_millis(),
         );
-        let mut initial_roots = Vec::new();
-        // Initialize the vectors
-        let mut s_secret_keys = vec![Scalar::zero(); num_transactions];
-        let mut s_old_values = vec![[BaseElement::ZERO; AFFINE_POINT_WIDTH + 2]; num_transactions];
-        let mut r_old_values = vec![[BaseElement::ZERO; AFFINE_POINT_WIDTH + 2]; num_transactions];
-        let mut s_indices = vec![0; num_transactions];
-        let mut r_indices = vec![0; num_transactions];
-        const EMPTY_PATH: Vec<Hash> = Vec::new();
-        let mut s_paths = vec![EMPTY_PATH; num_transactions];
-        let mut r_paths = vec![EMPTY_PATH; num_transactions];
-        let mut deltas = vec![BaseElement::ZERO; num_transactions];
 
         #[cfg(feature = "std")]
         let now = Instant::now();
-        // Repeat basic process for every transaction
+        // Fill in random receiver values in the tree
+        let mut new_accounts = 0;
```

Comment:
```suggestion
        #[cfg(feature = "std")]
        let mut new_accounts = 0;
```
It's used only for debugging